### DISCUSSION TITLE
How to subscribe to formState with useEffect?

### DISCUSSION THREAD

**Discussion by kuri-Maron** [react-hook-form/react-hook-form#4869] [Q&A]
I'm not good at English, so I'm sorry if I make a mistake.

## Question
I don't know the implementation to set the entire formState to an array of useEffect options, so please let me know.
Depending on the code sample, you may have set the entire formState or only a part of it, which is confusing.

## Background
I was looking for a way to perform a reset when submitting a form.
And I found the following document.
[SUBMIT WITH RESET](https://codesandbox.io/s/react-hook-form-handlesubmit-with-reset-xu1zu)
In this example, `isSubmitSuccessful` is specified in the optional array of useEffect.

<img width="923" alt="Cursor_と_useForm_-_reset___React_Hook_Form_-_Simple_React_forms_validation" src="https://user-images.githubusercontent.com/38233307/115313358-15611f00-a1ae-11eb-825c-9a9cad88bd63.png">

## formState documentation
When I looked at the docs, I interpreted that the entire formState should be specified as an optional array.
https://react-hook-form.com/api/useform/formstate

<img width="938" alt="Cursor_と_UseForm_-_FormState___React_Hook_Form_-_Simple_React_forms_validation" src="https://user-images.githubusercontent.com/38233307/115314650-9b7e6500-a1b0-11eb-9df7-0435054b4450.png">

I would appreciate it if you could tell me the difference from the above code.

**Comment by bluebill1049** [comment]
i have updated the CSB (will do doc after), the formState one is correct.

**Reply by bluebill1049** [reply]
`formState` should be the dep, not individual `state`
